Formula/krew.rb

```diff
@@ -15,15 +17,35 @@ class Krew < Formula
   depends_on "kubernetes-cli"
 
   def install
+    commit = Utils.popen_read("git", "rev-parse", "--short=8", "HEAD").chomp
     ENV["CGO_ENABLED"] = "0"
     # build in local dir to avoid this error:
     # go build: cannot write multiple packages to non-directory /usr/local/Cellar/krew/0.3.2/bin/krew
     mkdir "build"
-    system "go", "build", "-o", "build", "-tags", "netgo", "./cmd/krew/..."
+
+    ldflags = %W[
+      -w
+      -X sigs.k8s.io/krew/pkg/version.gitCommit=#{commit}
+      -X sigs.k8s.io/krew/pkg/version.gitTag=v#{version}
+    ]
+
+    system "go", "build", "-o", "build", "-tags", "netgo",
+      "-ldflags", ldflags.join(" "), "./cmd/krew"
     # install as kubectl-krew for kubectl to find as plugin
     bin.install "build/krew" => "kubectl-krew"
   end
 
+  def caveats
+    <<~EOS
+      To complete the installation, you need to add the following to your
+      ~/.bash_profile, ~/.bashrc or ~/.zshrc.
+
+        export PATH="${PATH}:${HOME}/.krew/bin"
+
+      Then, you can start using krew with the "kubectl krew" command.
+    EOS
+  end
+
   test do
     system "#{bin}/kubectl-krew", "version"
```

Comment:
This test should probably be improved.

We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
I can update this to `kubectl-krew update` (which does a `git clone` of some repo from the internet without creds). But it's not a better test in theory. To test an actual user journey, user must follow the caveat listed (manipulate `PATH`, and restart shell).

Reply:
You can manipulate the path variable in the test environment.

Reply:
Thanks, I've updated to exercise a real scenario (a kubectl plugin is installed via `krew` and its executable is symlinked from expected path).